crates/turbo-tasks-memory/src/task.rs

```diff
@@ -988,9 +1368,14 @@ impl Task {
                 log_scope_update!("removing initial scope");
                 let initial = backend.initial_scope;
                 if set.remove(initial) {
-                    self.remove_self_from_scope(&mut state, initial, backend, turbo_tasks);
                     let children = state.children.iter().copied().collect::<VecDeque<_>>();
-                    drop(state);
+                    self.remove_self_from_scope(
+                        &mut TaskMetaStateWriteGuard::Full(state),
+                        initial,
+                        backend,
+                        turbo_tasks,
+                    );
+                    // state ends here
```

Comment:
I'm not sure what this comment is meant to convey. Should state be dropped here?